src/jdk.httpserver/share/classes/sun/net/httpserver/Request.java

```diff
@@ -89,13 +112,25 @@ public String readLine() throws IOException {
                 } else {
                     gotCR = false;
                     consume(CR);
+                    if (firstClearRequest && offset == 0) {
+                        if (c < FIRST_CHAR) {
+                            throw new ProtocolException("Unexpected start of request line");
+                        }
+                        offset++;
```

Comment:
The changes look good to me. It took me a while to understand what this `offset` was and how those increments are used. It looks like it behaves merely like a boolean to decide whether or not to check the byte against the `FIRST_CHAR`. Maybe we could change `int offset` to `boolean firstByteChecked = false;`. It's also OK if you like to leave it in the current form.